gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -789,119 +795,261 @@ local function FindCorpse(ply)
+
+	net.Start("TTT2RevivalUpdate_IsReviving")
+	net.WriteBool(self.isReviving)
+	net.Send(self)
+end
+
+---
+-- Sets the blocking revival state.
+-- @param [default=false]boolean isBlockingRevival The blocking revival state
+-- @internal
+-- @realm server
+function plymeta:SetBlockingRevival(isBlockingRevival)
+	self.isBlockingRevival = isBlockingRevival or false
+
+	net.Start("TTT2RevivalUpdate_IsBlockingRevival")
+	net.WriteBool(self.isBlockingRevival)
+	net.Send(self)
+end
+
+---
+-- Sets the revival start time.
+-- @param [default=@{CurTime()}]number startTime The revival start time
+-- @internal
+-- @realm server
+function plymeta:SetRevivalStartTime(startTime)
+	self.revivalStartTime = startTime or CurTime()
+
+	net.Start("TTT2RevivalUpdate_RevivalStartTime")
+	net.WriteFloat(self.revivalStartTime)
+	net.Send(self)
+end
+
+---
+-- Sets the revival duration.
+-- @param [default=0.0]number duration The revival time
+-- @internal
+-- @realm server
+function plymeta:SetRevivalDuration(duration)
+	self.revivalDurarion = duration or 0.0
+
+	net.Start("TTT2RevivalUpdate_RevivalDuration")
+	net.WriteFloat(self.revivalDurarion)
+	net.Send(self)
+end
+
+---
+-- Sends a revival reason that is displayed in the clients revival HUD element.
+-- It supports a language identifier for translated strings.
+-- @param [default=nil]string name The text or the language identifer, nil to reset
+-- @param [opt]table params The params table used for @{LANG.GetParamTranslation}
+-- @realm server
+function plymeta:SendRevivalReason(name, params)
+	net.Start("TTT2SetRevivalReason")
+
+	if name then
+		net.WriteBool(false)
+		net.WriteString(name)
+
+		local paramsAmount = params and table.Count(params) or 0
+
+		net.WriteUInt(paramsAmount, 8)
+
+		if paramsAmount > 0 then
+			for k, v in pairs(params) do
+				net.WriteString(k)
+				net.WriteString(tostring(v))
+			end
+		end
+	else
+		net.WriteBool(true)
+	end
+
+	net.Send(self)
+end
+
+
+---
+-- Sets the last death position.
+-- @param Vector pos The death position
+-- @internal
+-- @realm server
+function plymeta:SetLastDeathPosition(pos)
+	self.lastDeathPosition = pos
+end
+
+---
+-- Sets the last spawn position.
+-- @param Vector pos The spawn position
+-- @internal
+-- @realm server
+function plymeta:SetLastSpawnPosition(pos)
+	self.lastSpawnPosition = pos
+end
+
+---
+-- Returns the last death position of the player.
+-- @return Vector The last death position
+-- @realm server
+function plymeta:GetDeathPosition()
+	return self.lastDeathPosition
+end
+
+---
+-- Returns the last spawn position of the player.
+-- @return Vector The last spawn position
+-- @realm server
+function plymeta:GetSpawnPosition()
+	return self.lastSpawnPosition
+end
+
+---
+-- Sets the if a player was active (TEAM_TERROR) in a round.
+-- @param boolean state The state
+-- @internal
+-- @realm server
+function plymeta:SetActiveInRound(state)
+	if state and (GetRoundState() ~= ROUND_ACTIVE or not self:IsTerror()) then return end
```

Comment:
Is this check really needed? I mean if you wanna set it, you don't get feedback whether it was successfully.

Reply:
I mean, not necessarily. But I though it would make the function safer.

Reply:
I think I'm with @Alf21, as this is unexpected behavior, and silently returning, makes debugging hard and logical mistakes even harder to find.